RPM Spec fix

jmstover opened the pull request:
With #2261 , we need some minor tweaks to the spec to build an RPM.

With this, and #2261 the following will work:
```
./mconfig && \
make -C builddir/ dist && \
rpmbuild -tb singularity-3.0.0.tar.gz
```

Or simpler:
```
./mconfig && \
make -C builddir/ rpm
```
**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)


Attn: @singularity-maintainers

jmstover committed:
RPM Spec fix

With PR2261, we need some minor tweaks to the spec to build an RPM.

With this, and 2261 the following will work:

./mconfig && \
make -C builddir/ dist && \
rpmbuild -tb singularity-3.0.0.tar.gz

jmstover committed:
Remove %setup, and move steps around

jmstover committed:
Modify rpm build target

DrDaveD commented on mlocal/frags/Makefile.stub:
```diff
@@ -272,9 +272,9 @@ test:
+rpm: dist
 	@echo " BUILD RPM"
-	$(V)cd $(SOURCEDIR) && dist/rpm/buildrpm
+	$(V)cd $(SOURCEDIR) && rpmbuild -tb singularity-*.tar.gz
```
Could this use a specific version number instead of '*', in case there is another tarball left from something?

jmstover replied:
Yeah, we're currently working on getting a quick PR that's passing just the version number (i.e. 3.0.0), into the Makefile for use.

And a question for you...  This is the Travis error:

```
 DIST create tarball: singularity-3.0.0.tar.gz
*** Please tell me who you are.
Run
  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"
to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: empty ident  <testuser@travis-job-2759d569-49a5-49c8-87c0-a56463bcacef.(none)> not allowed
```

Would you know how to uhh... fix the Travis config here?

DrDaveD replied:
It looks like you've gotten past that, it's now failing because it can't find go.  Unfortunately I don't have time to look at it this morning, I could this afternoon (Chicago time) if it is still not working.

jmstover replied:
No... the `go` error is misleading at this point. The deps action is getting triggered when we make the tar ball.

What I did for the moment is Hard Code the Tarball name at that point. Does the `-V 0.0` _need_ to be used here (or is it needed anymore now that we include a VERSION file)? I'm thinking this is an interaction between the `./mconfig` run saying the version is 0.0, and other parts that are just looking at the output of `git describe` for what it thinks the version should be.

So we have some disjointedness with a specified Version, and what the tarball believes it's version should actually be.

DrDaveD replied:
The only reason for the -V 0.0 was that some version was needed, and there wasn't any.  If you've now got a VERSION file you can remove the -V 0.0 from .travis/rpmbuild_test

jmstover committed:
Test RPM build fix

jmstover committed:
Merge branch 'master' of github.com:sylabs/singularity into spec_fix

jmstover committed:
Use SHORT_VERSION in tar.gz name

jmstover committed:
Update make-dist.sh

git 1.7.1 on RHEL 6 doesn't support `--format=tar.gz`. So, we'll
split it and use `--format=tar`, then gzip the resulting file.

jmstover committed:
Add some debug help

jmstover committed:
Fetch repository

jmstover committed:
Move fetch line

jmstover committed:
Test hard coding version

jmstover committed:
Clean up versioning

jmstover committed:
Change version naming

Change naming to be more consistent with mconfig naming.

jmstover committed:
Some debugging

jmstover committed:
Add git in as a package

jmstover committed:
More debugging info

jmstover committed:
More debugging

jmstover committed:
Escape quotes

jmstover committed:
Update make-dist.sh script

jmstover committed:
Debugging fun...

jmstover committed:
Try rewrite to use heredoc

jmstover committed:
rpmbuild_test

* Revert heredoc test
* Remove `/singularity` from BLD line

jmstover committed:
Add in a singularity-[ver] root in BUILD that is used

DrDaveD commented on .travis/rpmbuild_test:
```diff
@@ -8,21 +8,30 @@ yum install -y openssl-devel libuuid-devel
   make -C builddir rpm
   sudo yum install -y $HOME/rpmbuild/RPMS/*/*.rpm
-  BLD=`echo $HOME/rpmbuild/BUILD/singularity-*`
+  BLD=`echo $HOME/rpmbuild/BUILD`
```
No, it's important for the singularity-* level to be there.  When people run rpmbuild the rpmbuild/BUILD directory is shared, and packages avoid conflicting with each other by creating a top-level directory of package-version.  This can be fixed by easily added by adding “mkdir %{name}-%{version}” and “cd %{name}-%{version}" to the beginning of the %build step in singularity.spec.in and “cd %{name}-%{version}" to the beginning of "%install", a better solution.

yhcote reviewed (approved):
Looks good to me!

DrDaveD commented on .travis/rpmbuild_test:
```diff
@@ -8,21 +8,30 @@ yum install -y openssl-devel libuuid-devel
+  set -e
+  ./mconfig 
   sudo yum-builddep -y dist/rpm/singularity.spec
+  git fetch --all
```
I don't believe this fetch does anything, better to leave it out.

jmstover replied:
:+1:

DrDaveD commented on .travis/rpmbuild_test:
```diff
@@ -8,21 +8,30 @@ yum install -y openssl-devel libuuid-devel
   export GOROOT=$BLD/go
   export GOPATH=$BLD/gopath
   PATH=$GOROOT/bin:$GOPATH/bin:$PATH
+  echo " BUILD DIR "
```
What's this echo for?  A leftover debug statement?

jmstover committed:
Remove git fetch

jmstover committed:
Really get rid of debug output... I hope

yhcote committed:
Merge branch 'master' into spec_fix
